portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,35 +14,97 @@
 
 package com.google.sps.servlets;
 
+import java.util.Map;
+import java.util.HashMap;
 import com.google.gson.Gson;
-import com.google.common.collect.ImmutableList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns inputted fun facts from users and stores previously inputted facts. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final ImmutableList<String> FRUIT_FACTS =
-    ImmutableList.of(
-      "A strawberry is not an actual berry", 
-      "Kiwis contain more Vitamin C than oranges", 
-      "The pineapple is actually a berry");
+  private static final String FIRST_NAME_KEY = "first-name";
+  private static final String LAST_NAME_KEY = "last-name";
+  private static final String FUN_FACT_KEY = "user-fun-fact";
+  private static final String NO_NAME = "Anonymous";
+  private Map<String, String> oldFacts = new HashMap<String, String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String fruitFactsJson = convertToJsonUsingGson(FRUIT_FACTS);
+    String oldFactsJson = convertToJsonUsingGson(oldFacts);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
-    response.getWriter().println(fruitFactsJson);
+    response.getWriter().println(oldFactsJson);
   }
 
-  private String convertToJsonUsingGson(ImmutableList<String> fruitFacts) {
+  private static String convertToJsonUsingGson(Map<String, String> oldFacts) {
     Gson gson = new Gson();
-    String json = gson.toJson(fruitFacts);
+    String json = gson.toJson(oldFacts);
     return json;
   }
-}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String userFirstName = getParameter(request, /*content=*/ FIRST_NAME_KEY, /*defaultValue=*/ "");
+    String userLastName = getParameter(request, /*content=*/ LAST_NAME_KEY, /*defaultValue=*/ "");
+    String userFunFact = getParameter(request, /*content=*/ FUN_FACT_KEY, /*defaultValue=*/ "");
+
+    // No need to change oldFacts if there is no new fact
+    if (userFunFact.isEmpty()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    // Define key and value pair to add to oldFacts
+    String factKey = getFactKey(userFirstName, userLastName);
+    String factValue = getFactValue(oldFacts, userFunFact, factKey);
+    
+    // Store user's input to access later
+    oldFacts.put(factKey, factValue);
```

Comment:
Just a note that if two different people happen to have the same name, their facts will be combined.